File: Brute_Force/javascript/10819.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const n = Number(input[0]);
const nums = input[1].split(" ").map(Number);

const cal = (arr) => {
  let total = 0;
  for (let i = 0; i < arr.length - 1; i++) {
    total += Math.abs(arr[i] - arr[i + 1]);
  }
  return total;
};
const permutation = (nums, r) => {
  const visited = new Array(n).fill(false);
  const selected = [];
  const generate = (arr, result) => {
    if (arr.length === r) {
      return cal(arr);
    }
    for (let i = 0; i < nums.length; i++) {
      if (visited[i]) continue;
      visited[i] = true;
      arr.push(nums[i]);
      result = Math.max(result, generate(arr, result));
      arr.pop();
      visited[i] = false;
    }
    return result;
  };
  return generate(selected, 0);
};
console.log(permutation(nums, n));

File: Brute_Force/python/10819.py
n = int(input())
nums = list(map(int,input().split()))
def cal(arr):
  total = 0
  for i in range(len(arr)-1):
    total += abs(arr[i] - arr[i+1])
  return total

def permutation(arr,r):
  result = 0
  arr = sorted(arr)
  visited = [0 for _ in range(len(arr))]
  def generate(selected,visited,result):
    if len(selected) == r:
      return cal(selected)
    for i in range(len(arr)):
      if not visited[i]:
        selected.append(arr[i])
        visited[i] = 1
        result = max(result,generate(selected,visited,result))
        visited[i] = 0
        selected.pop()
    return result
  result = generate([],visited,result)
  return result
print(permutation(nums,n))